components/LoginButton.tsx

```diff
@@ -0,0 +1,19 @@
+import { useSession, signIn, signOut } from 'next-auth/react'
+
+export default function LoginButton() {
+  const { data: session } = useSession()
+  if (session) {
+    return (
+      <>
+        Signed in as {session.user?.email} <br />
+        <button onClick={() => signOut()}>Sign out</button>
+      </>
+    )
+  }
+  return (
+    <>
+      Not signed in <br />
+      <button onClick={() => signIn()}>Sign in</button>
+    </>
+  )
+}
```

Comment:
nit: 

Just a proposal: 

```suggestion
   const onClick = session ? signOut : signIn;
   
   const captionMessage = session ? `Signed in as ${session.user?.email}` : "Not signed in";
   
   const buttonMessage = session ? "Sign out" : "Sign in";

    return (
      <>
        {captionMessage} <br />
        <button onClick={onClick}>{buttonMessage}</button>
      </>
    )
}
```

Reply:
Improved as you suggested